src/test/java/seedu/address/logic/commands/RequiredCommandTest.java

```diff
@@ -0,0 +1,162 @@
+    }
+
+    @Test
+    public void validGetInternshipStorageTest() {
+        Optional<ReadOnlyGradPad> empty = requiredCommand.getInternshipStorage();
+        assertEquals(null, empty);
+    }
+
+    @Test
+    public void setFoundationStorageValidPath_returnsFilledOptional() throws IOException, DataConversionException {
+        JsonGradPadStorage expectedJsonStorage = new JsonGradPadStorage(FOUNDATION_PATH);
+        Optional<ReadOnlyGradPad> expected = expectedJsonStorage.readGradPad();
+        requiredCommand.setFoundationStorage(FOUNDATION_PATH);
+        Optional<ReadOnlyGradPad> actual = requiredCommand.getFoundationStorage();
+        assertEquals(actual, expected);
+    }
+    @Test
+    public void setITprofStorageValidPath_returnsFilledOptional() throws IOException, DataConversionException {
+        JsonGradPadStorage expectedJsonStorage = new JsonGradPadStorage(ITPROF_PATH);
+        Optional<ReadOnlyGradPad> expected = expectedJsonStorage.readGradPad();
+        requiredCommand.setITprofStorage(ITPROF_PATH);
+        Optional<ReadOnlyGradPad> actual = requiredCommand.getITprofStorage();
+        assertEquals(actual, expected);
+    }
+    @Test
+    public void setMathAndScienceStorageValidPath_returnsFilledOptional() throws IOException, DataConversionException {
+        JsonGradPadStorage expectedJsonStorage = new JsonGradPadStorage(MATHANDSCI_PATH);
+        Optional<ReadOnlyGradPad> expected = expectedJsonStorage.readGradPad();
+        requiredCommand.setMathAndScienceStorage(MATHANDSCI_PATH);
+        Optional<ReadOnlyGradPad> actual = requiredCommand.getMathAndScienceStorage();
+        assertEquals(actual, expected);
+    }
+    @Test
+    public void setScienceStorageValidPath_returnsFilledOptional() throws IOException, DataConversionException {
+        JsonGradPadStorage expectedJsonStorage = new JsonGradPadStorage(SCIENCE_PATH);
+        Optional<ReadOnlyGradPad> expected = expectedJsonStorage.readGradPad();
+        requiredCommand.setScienceStorage(SCIENCE_PATH);
+        Optional<ReadOnlyGradPad> actual = requiredCommand.getScienceStorage();
+        assertEquals(actual, expected);
+    }
+    @Test
+    public void setInternshipStorageValidPath_returnsFilledOptional() throws IOException, DataConversionException {
+        JsonGradPadStorage expectedJsonStorage = new JsonGradPadStorage(INTERNSHIP_PATH);
+        Optional<ReadOnlyGradPad> expected = expectedJsonStorage.readGradPad();
+        requiredCommand.setInternshipStorage(INTERNSHIP_PATH);
+        Optional<ReadOnlyGradPad> actual = requiredCommand.getInternshipStorage();
+        assertEquals(actual, expected);
+    }
+
+    @Test
+    public void setFoundationStorageInvalidPath_returnsEmptyOptional() throws IOException, DataConversionException {
+        requiredCommand.setFoundationStorage(INVALID_PATH);
+        Optional<ReadOnlyGradPad> actual = requiredCommand.getFoundationStorage();
+        assertEquals(actual, Optional.empty());
```

Comment:
Perhaps we can consider using this instead:

```suggestion
        assertTrue(actual.isEmpty());
```

As mentioned in the java API, it's generally not recommended to do equality checks for Optionals cos they produce unpredictable results 😁  Same applies for the other tests below and also `ScienceCommandTest`

Reply:
okay roger doger!